### PR TITLE
Dropdown question rendering

### LINKED ISSUE #6923: [North Star Applicant UI] Render dropdown questions
#### Describe the solution presented in this issue.
Render dropdown questions in Thymeleaf

#### Done when:
What criteria need to be met for the issue to be finished? 
- [ ] Users can see and respond to dropdown questions, including handling validation errors

### PR DESCRIPTION
**Pull request by greg-hyde**
### Description

Render dropdown questions in Thymeleaf

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

#### User visible changes

- [x] Didn't use a message in applicant facing code that isn't translated yet (unless behind a flag)
- [x] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [x] Manually evaluated tab order

#### New Features

- [x] Conditioned new functionality on a [FeatureFlag](https://github.com/civiform/civiform/wiki/Feature-Flags)
- [x] Wrote browser tests with the feature flag off and on, etc.

### Issue(s) this completes

Fixes #6923

### CODE REVIEW

**Comment by greg-hyde** [comment]
* **#7005** <a href="https://app.graphite.dev/github/pr/civiform/civiform/7005?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @greg-hyde and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: browser-test/src/applicant/questions/dropdown.test.ts [hunk 1]
```diff
@@ -181,4 +164,63 @@ test.describe('Dropdown question for applicant flow', () => {
       await validateAccessibility(page)
     })
   })
+
+  test.describe('single dropdown questio with north star flag enabled', () => {
```
Comment (gwendolyngoetz) [inline browser-test/src/applicant/questions/dropdown.test.ts:168]:
```suggestion
  test.describe('single dropdown question with north star flag enabled', () => {
```

File: server/app/views/questiontypes/DropdownQuestionFragment.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    <div th:replace="~{questiontypes/QuestionBaseFragment :: validationErrors(${singleSelectQuestion.getValidationErrors().get(question.getContextualizedPath())})}"></div>
+    
+    <select class="usa-select"
+        name="options"
+        th:id="${inputId}"
+        th:name="${singleSelectQuestion.getSelectionPath()}"
+        th:aria-required="${!question.isOptional()}"
+        th:aria-describedby="${questionId} + '-description'">
+      <option value hidden th:selected="${!singleSelectQuestion.hasValue()}" th:text="#{placeholder.noDropdownSelection}"></option>
+      <optgroup>
```
Comment (gwendolyngoetz) [inline server/app/views/questiontypes/DropdownQuestionFragment.html]: question:  Do we need this `optgroup`?

File: server/app/views/questiontypes/DropdownQuestionFragment.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<div th:fragment="dropdown(question)"
+     th:with="singleSelectQuestion=${question.createSingleSelectQuestion()},
+              questionId=${#strings.randomAlphanumeric(8)},
+              inputId=${#strings.randomAlphanumeric(8)}"
+     th:id="${questionId}">
+    <!-- Title and Help Text -->
+    <div th:replace="~{questiontypes/QuestionBaseFragment :: titleAndHelpTextSingleInput(${question}, ${questionId}, ${inputId})}"></div>
```
Comment (gwendolyngoetz) [inline server/app/views/questiontypes/DropdownQuestionFragment.html]: question: Should this be a label?

**Review by gwendolyngoetz — CHANGES_REQUESTED** [review]

File: server/app/views/questiontypes/DropdownQuestionFragment.html (hunk 3 above)
Comment (greg-hyde) [reply server/app/views/questiontypes/DropdownQuestionFragment.html]: It doesn't really matter. we're using th:replace, which removes the <div> I'm adding and replaces it with the <label> rendered by that fragment. I've been using <div> in these cases but not for any particular reason.

File: server/app/views/questiontypes/DropdownQuestionFragment.html (hunk 2 above)
Comment (greg-hyde) [reply server/app/views/questiontypes/DropdownQuestionFragment.html]: 👍 removed

**Review by gwendolyngoetz — APPROVED** [review]

**Pull request merged**